src/coreComponents/constitutive/solid/TriaxialDriver.cpp

```diff
@@ -0,0 +1,309 @@
+  // storing constitutive data
+
+  conduit::Node node;
+  dataRepository::Group rootGroup( "root", node );
+  dataRepository::Group discretization( "discretization", &rootGroup );
+
+  discretization.resize( 1 );   // one element
+  baseSolid.allocateConstitutiveData( discretization, 1 );   // one quadrature point
+
+  // set the initial stress state using the data table
+
+  arrayView3d< real64, solid::STRESS_USD > stressArray = baseSolid.getStress();
+
+  stressArray( 0, 0, 0 ) = m_table( 0, SIG0 );
+  stressArray( 0, 0, 1 ) = m_table( 0, SIG1 );
+  stressArray( 0, 0, 2 ) = m_table( 0, SIG2 );
+
+  baseSolid.saveConvergedState();
+
+  // pass the solid through the ConstitutivePassThru to downcast from the
+  // base type to a known model type.  the lambda here then executes the
+  // appropriate test driver. note that these calls will move data to device if available.
+
+  ConstitutivePassThru< SolidBase >::execute( baseSolid, [&]( auto & selectedSolid )
+  {
+    using SOLID_TYPE = TYPEOFREF( selectedSolid );
+
+    if( m_mode == "triaxial" )
+    {
+      runMixedControlTest< SOLID_TYPE >( selectedSolid, m_table );
+    }
+    else if( m_mode == "volumetric" || m_mode == "oedometer" )
+    {
+      runStrainControlTest< SOLID_TYPE >( selectedSolid, m_table );
+    }
+  } );
+
+  // move table back to host for output
+  m_table.move( LvArray::MemorySpace::CPU );
+
+  if( m_outputFile != "none" )
+  {
+    outputResults();
+  }
+
+  if( m_baselineFile != "none" )
+  {
+    compareWithBaseline();
+  }
+
+  return false;
+}
+
+
+void TriaxialDriver::outputResults()
+{
+  FILE * fp = fopen( m_outputFile.c_str(), "w" );
+
+  fprintf( fp, "# column 1 = time\n" );
+  fprintf( fp, "# column 2 = axial_strain\n" );
+  fprintf( fp, "# column 3 = radial_strain_1\n" );
+  fprintf( fp, "# column 4 = radial_strain_2\n" );
+  fprintf( fp, "# column 5 = axial_stress\n" );
+  fprintf( fp, "# column 6 = radial_stress_1\n" );
+  fprintf( fp, "# column 7 = radial_stress_2\n" );
+  fprintf( fp, "# column 8 = newton_iter\n" );
+  fprintf( fp, "# column 9 = residual_norm\n" );
+
+  for( localIndex n=0; n<=m_numSteps; ++n )
+  {
+    for( localIndex col=0; col<m_numColumns; ++col )
+    {
+      fprintf( fp, "%.4e ", m_table( n, col ) );
+    }
+    fprintf( fp, "\n" );
+  }
+  fclose( fp );
+}
+
+
+void TriaxialDriver::compareWithBaseline()
+{
+  // open baseline file
+
+  std::ifstream file( m_baselineFile.c_str() );
+  GEOSX_THROW_IF( !file.is_open(), "Can't seem to open the baseline file " << m_baselineFile, InputError );
+
+  // discard file header
+
+  string line;
+  for( localIndex row=0; row < m_numColumns; ++row )
+  {
+    getline( file, line );
+  }
+
+  // read data block.  we assume the file size is consistent with m_table,
+  // but check for a premature end-of-file. we then compare results value by value.
+  // we ignore the newton iteration and residual columns, as those may be platform
+  // specific.
+
+  real64 value;
```

Comment:
const?

Reply:
non-const.  this is the buffer the file values are read into.